### DISCUSSION TITLE
Proprietary Playlist File

### DISCUSSION THREAD

**Discussion by TaxicabMessiah** [zaps166/QMPlay2#766] [Ideas]
Exporting internet radio playlists leaves me with a .qmplay2radio file.  This doesn't translate to any other player and is an unrecognizable file, of course.  Can't the format be .csv or .m3u or .pls?  Anyway, truly awesome media player.  It's what Foobar2000 always wanted to be but couldn't.  I noticed the quick playlist reference trick when in full screen spectrum mode too...very slick.

**Comment by zaps166** [comment]
Hi. `qmplay2radio` format was the simplest way to allow users to backup their "My radio stations", because this file contains icons, too.

I'll look at this and try to replace it to `.pls` format. I hope it can handle icons without breaking compatibility with other players.

**Reply by TaxicabMessiah** [reply]
Actually I was wrong.  There is a drop-down menu when I save the file that I missed.  It's available to save as .m3u8 and all the other relevant formats.  The only bug I'm running into so far is that movies in full screen mode flicker when you move to the bottom to reveal the seek bar.

**Reply by zaps166** [reply]
Aaa yes, forgot, but doesn't store icons.

**Reply by zaps166** [reply]
> The only bug I'm running into so far is that movies in full screen mode flicker when you move to the bottom to reveal the seek bar.

Windows? It depends on drivers. Do you use Vulkan or OpenGL?

**Reply by zaps166** [reply]
You can disable this option to prevent flickering, but it will lower the GPU performance (higher power consumption) + you can have micro-stuttering due to desync with DWM compositor.

First try to update GPU drivers. On my Radeon on PC I don't have flickering since some driver updates (only stuttering visible when using bottom bar, but have to verify again - I use Linux all the time). It depends on drivers - the worst on Intel+Nvidia laptops...

![Zrzut ekranu_20241020_220928](https://github.com/user-attachments/assets/a2227923-95e6-4de2-9a72-d53c09172bd0)

**Reply by TaxicabMessiah** [reply]
It defaulted to Vulkan.  I just switched to OpenGL and everything now works perfectly.  (Intel mobile/igpu on Windows 11).  Thanks!

**Reply by zaps166** [reply]
OpenGL is bad idea, stay with Vulkan.

**Reply by zaps166** [reply]
I don't remember why, but in OpenGL I have bypass compositor disabled by default, so that's why it doesn't flicker. by default.

If you need to not flicker, better stay with Vulkan and disable it there as in screenshot.

**Reply by TaxicabMessiah** [reply]
"bypass compositor in full screen" was checked by default on Vulkan and still had the flicker issue but when unchecked the problem is solved as well.  Interesting.

**Reply by zaps166** [reply]
bypassing compositor gets exclusive full screen for QMPlay2 (like games has on full screen). On some old/bugged GPU drivers it flickers on Windows, unfortunately.

**Reply by TaxicabMessiah** [reply]
Well it's the sickest media player I've ever found.  Thx.  Only other things, since your here, would be that
- the right-most third of the rainbow visualizer is inactive for music
- quick toggle for audio/subtitles like MPV has on screen would be useful instead of right clicking through the menu like VLC
That's literally all I can think of for complaints/improvements.  I can't even think of another media player that defaults to video playback of internet radio stations with video available.  Foobar can't do that, not AIMP nor anything else.

**Reply by zaps166** [reply]
> the right-most third of the rainbow visualizer is inactive for music

Do you mean high frequencies? Maybe your music doesn't have high frequency. Try FLAC or AudioCD or other high-quality music without losy codecs and low pass filtrers.

If your music has 44100 Hz sample rate it means it can have [0-22050) Hz frequencies (see: Nyquist frequency). Music usually is cut to ~20000 Hz or cut by lossy codecs (low quality music) to e.g. 16000Hz or lower (e.g. YouTube). If you can't see anything there it means there are no such frequencies in your music or you use Equalizer to cut high frequencies in QMPlay2 or have custom samplerate enables in QMPlay2 playback settings.

> quick toggle for audio/subtitles like MPV has on screen would be useful instead of right clicking through the menu like VLC

I'll think about it, good idea anyway.

**Reply by TaxicabMessiah** [reply]
Oh yeah I'm using lots of HQ flac files like 192 and 24 bit.  The visualizer just doesn't match the audio for some reason.  Here is a high-res flute file and it's all low end bars.  I couldn't find any settings to tweak.
![Screenshot 2024-10-20 174657](https://github.com/user-attachments/assets/5210e07f-b24d-4b9b-adc0-71dd68dc9360)

**Reply by zaps166** [reply]
Because you have 192kHz samplerate (possible 96 kHz audio frequencies), but audio is usually up to 20kHz, so that's why.

In 192kHz samplerate point of view, 20kHz is low frequency :sweat_smile: 

I can add scaling option anyway.

**Reply by TaxicabMessiah** [reply]
Thats funny, you're right.  Here is a low-res 311 file and its a rainbow cornucopia
![Screenshot 2024-10-20 180837](https://github.com/user-attachments/assets/93c45be3-4de2-4806-89aa-1b27d1f12f1f)

**Reply by zaps166** [reply]
>  I couldn't find any settings to tweak.

Added, will be in next release.

**Reply by TaxicabMessiah** [reply]
Sick.  I've been telling people about this player.  Is this a fork of Foobar or another player?  Why is it so polished and yet so obscure.

**Reply by zaps166** [reply]
> Is this a fork of Foobar or another player?

Where do you see Foobar? It's different program.

> Why is it so polished and yet so obscure.

Could you explain? Just don't tell me settings window is messy - I know it :smile:

**Reply by TaxicabMessiah** [reply]
I'm saying QMPlay2 is very very good.  It's soo well rounded and yet no one knows about it.  Is this program a fork of another program like Foobar?

**Reply by zaps166** [reply]
> I'm saying QMPlay2 is very very good.

Thanks!

>  Is this program a fork of another program like Foobar?

No, it's another program written from scratch.

**Reply by zaps166** [reply]
I hope FFT spectrum is working better now with the new release! Also you should have lower CPU usage when using wallpaper with visualizations.

**Reply by TaxicabMessiah** [reply]
This is very cool.  I am now getting the background of the equalizer settings misplaced though.  Meaning if I use custom wallpaper the equalizer takes a random part of the image to use for its background instead of the part of the image that would go there naturally.  CPU usage tweak is sick...it was already faster than foobar or any other player with this much horsepower.  Nothing opens faster.

**Reply by zaps166** [reply]
I don't know foobar - does it have large visualizations?

Refreshing visualizations ~60 FPS with gradient + repainting wallpaper there in high res monitor is a thing to the CPU. Minimize window and check CPU usage again.

>  I am now getting the background of the equalizer settings misplaced though. 

I didn't touch equalizer. On Linux I see it also in older QMPlay2 releases. Maybe Qt update did this or you set different Qt style which triggered this bug? However I don't know why this happens sometimes (Qt should detect this scenario and don't perform scrolling optimization).

**Reply by TaxicabMessiah** [reply]
I only use portable version if available.  I'm on windows and I originally tried the QT version of portable which gave me that issue.  But now I've just tried the non-QT version of portable and it's working just fine.  BTW I also added some other suggestions in the discussion page for your long term consideration.  I'm going to work on setting up the updated player tonight, thanks!

**Reply by zaps166** [reply]
> BTW I also added some other suggestions in the discussion page for your long term consideration.

Yes, I noticed.

>  I'm on windows and I originally tried the QT version of portable which gave me that issue. But now I've just tried the non-QT version of portable and it's working just fine.

What? Where you have "non-QT version" of QMPlay2 :grin: ? QMPlay2 is using Qt for almost everything. What package do you exactly use and where exactly the Equalizer wallpaper issue exist?

**Reply by TaxicabMessiah** [reply]
Right here.
![Screenshot 2024-12-07 184444](https://github.com/user-attachments/assets/0567f542-c854-47e2-b68a-c5d94b5bbf89)

**Reply by zaps166** [reply]
Both are Qt. One is Qt5, another Qt6. Qt5 is for Windows 7 and 8 compatibility. Which is working better for you with equalizer?

**Reply by TaxicabMessiah** [reply]
I see, yes QT6 (the unlabeled file) seems to work just fine.
